src/common/helpers.js

```diff
@@ -17,33 +18,53 @@ export const debounce = (func, wait) => {
   return innerFunc;
 };
 
-// Returns a subset of the given object including only the given keys, with values optionally replaced by a fn.
+/** Returns true if propName is a non-null, defined property of the props object (can be any object, not just React props). */
+export const propExists = (props, propName) =>
+  props && props.hasOwnProperty(propName) && props[propName] != null;
+
+/** Given two objects (props and state), returns the value of propName from props if present, or from state otherwise. */
+export const propOrState = (props, state, propName) =>
+  propExists(props, propName) ? props[propName] : state[propName];
+
+/** Returns a subset of the given object including only the given keys, with values optionally replaced by a fn. */
 export const selectKeys = (obj, keys, fn = val => val) =>
   keys.reduce((values, key) => ({ ...values, [key]: fn(obj[key]) }), {});
 
-// Returns a subset of the given object with a validator function applied to its keys.
+/** Returns a copy of the object with all its values replaced by the given function which takes a key.  */
+export const mapKeys = (obj, fn = key => obj[key]) =>
```

Comment:
I believe this is no longer used now that stateKnobs has been removed.

Reply:
+1